README.md

```diff
@@ -338,6 +338,7 @@ coupled as possible. New features should typically be implemented as new modules
 | [org.librarysimplified.content.api](simplified-content-api)                                   |Content resolver API|
 | [org.librarysimplified.crashlytics](simplified-crashlytics)                                   |Crashlytics|
 | [org.librarysimplified.crashlytics.api](simplified-crashlytics-api)                           |Crashlytics functionality|
+| [org.librarysimplified.deeplinks.controller.api](simplified-deeplinks-controller-api)         |Documents|
```

Comment:
Will update README and remove the fully qualified class names - yes, that must be an artifact of Android Studio refactoring when I moved the classes into their own module, sorry about that! Will make these changes and tag you when they are done.

Reply:
I've been getting this error since yesterday and trying to figure out what is causing it:

```
Type org.nypl.simplified.profiles.controller.api.BuildConfig is defined multiple times: /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-profiles-controller-api/build/.transforms/5aa75ff6087c1776d5bf80b44ea7011c/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex, /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-deeplinks-controller-api/build/.transforms/0863ec246ae04abdc1f2d3f172457ab7/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex
```

Trying to figure out if that was from a code change or if it's a current state of my environment - will investigate.

Reply:
Strange, I am now getting new errors and all I did was clean and build again.

They are:
```
java.lang.AssertionError: symbolic reference class is not accessible: class sun.nio.ch.FileChannelImpl, from class org.jetbrains.kotlin.com.intellij.util.io.FileChannelUtil (unnamed module @16d04d3d)
Caused by: java.lang.AssertionError: symbolic reference class is not accessible: class sun.nio.ch.FileChannelImpl, from class org.jetbrains.kotlin.com.intellij.util.io.FileChannelUtil (unnamed module @16d04d3d)
```

And:

```
java.lang.IllegalAccessException: symbolic reference class is not accessible: class sun.nio.ch.FileChannelImpl, from class org.jetbrains.kotlin.com.intellij.util.io.FileChannelUtil (unnamed module @16d04d3d)
Caused by: java.lang.IllegalAccessException: symbolic reference class is not accessible: class sun.nio.ch.FileChannelImpl, from class org.jetbrains.kotlin.com.intellij.util.io.FileChannelUtil (unnamed module @16d04d3d)
```

Ah, it does seem to be the same root cause of the BuildConfig being defined multiple times, though:

```

AGPBI: {"kind":"error","text":"Type org.nypl.simplified.profiles.controller.api.BuildConfig is defined multiple times: /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-profiles-controller-api/build/.transforms/5aa75ff6087c1776d5bf80b44ea7011c/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex, /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-deeplinks-controller-api/build/.transforms/0863ec246ae04abdc1f2d3f172457ab7/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex","sources":[{"file":"/Users/jdempcy/Development/ThePalaceProject/android-core/simplified-profiles-controller-api/build/.transforms/5aa75ff6087c1776d5bf80b44ea7011c/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex"}],"tool":"D8"}
com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
Type org.nypl.simplified.profiles.controller.api.BuildConfig is defined multiple times: /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-profiles-controller-api/build/.transforms/5aa75ff6087c1776d5bf80b44ea7011c/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex, /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-deeplinks-controller-api/build/.transforms/0863ec246ae04abdc1f2d3f172457ab7/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex
	at com.android.builder.dexing.D8DexArchiveMerger.getExceptionToRethrow(D8DexArchiveMerger.java:151)
	at com.android.builder.dexing.D8DexArchiveMerger.mergeDexArchives(D8DexArchiveMerger.java:138)
	at com.android.build.gradle.internal.tasks.DexMergingWorkAction.merge(DexMergingTask.kt:863)
	at com.android.build.gradle.internal.tasks.DexMergingWorkAction.run(DexMergingTask.kt:809)
	at com.android.build.gradle.internal.profile.ProfileAwareWorkAction.execute(ProfileAwareWorkAction.kt:74)
	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
	at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:66)
	at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:62)
	at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:97)
	at org.gradle.workers.internal.NoIsolationWorkerFactory$1.lambda$execute$0(NoIsolationWorkerFactory.java:62)
	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:79)
	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:79)
	at org.gradle.workers.internal.AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
	at org.gradle.workers.internal.NoIsolationWorkerFactory$1.execute(NoIsolationWorkerFactory.java:59)
	at org.gradle.workers.internal.DefaultWorkerExecutor.lambda$submitWork$2(DefaultWorkerExecutor.java:206)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:214)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:164)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:131)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:61)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: com.android.tools.r8.CompilationFailedException: Compilation failed to complete, origin: /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-profiles-controller-api/build/.transforms/5aa75ff6087c1776d5bf80b44ea7011c/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex
	at Version.fakeStackEntry(Version_3.0.76.java:0)
	at com.android.tools.r8.utils.w.a(SourceFile:68)
	at com.android.tools.r8.utils.w.a(SourceFile:28)
	at com.android.tools.r8.utils.w.a(SourceFile:27)
	at com.android.tools.r8.utils.w.b(SourceFile:3)
	at com.android.tools.r8.D8.run(D8.java:11)
	at com.android.builder.dexing.D8DexArchiveMerger.mergeDexArchives(D8DexArchiveMerger.java:136)
	... 35 more
Caused by: com.android.tools.r8.internal.b: Type org.nypl.simplified.profiles.controller.api.BuildConfig is defined multiple times: /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-profiles-controller-api/build/.transforms/5aa75ff6087c1776d5bf80b44ea7011c/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex, /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-deeplinks-controller-api/build/.transforms/0863ec246ae04abdc1f2d3f172457ab7/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex
	at com.android.tools.r8.internal.UA.a(SourceFile:14)
	at com.android.tools.r8.internal.UA.a(SourceFile:22)
	at com.android.tools.r8.internal.Fw.b(SourceFile:6)
	at com.android.tools.r8.internal.Fw.a(SourceFile:23)
	at com.android.tools.r8.internal.Fw.a(SourceFile:10)
	at java.base/java.util.concurrent.ConcurrentHashMap.merge(Unknown Source)
	at com.android.tools.r8.internal.Fw.a(SourceFile:6)
	at com.android.tools.r8.graph.I0$c.f(SourceFile:3)
	at com.android.tools.r8.dex.a.a(SourceFile:94)
	at com.android.tools.r8.dex.a.a(SourceFile:23)
	at com.android.tools.r8.D8.a(D8.java:21)
	at com.android.tools.r8.D8.d(D8.java:11)
	at com.android.tools.r8.D8.b(D8.java:1)
	at com.android.tools.r8.utils.w.a(SourceFile:24)
	... 38 more

Caused by: com.android.tools.r8.CompilationFailedException: Compilation failed to complete, origin: /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-profiles-controller-api/build/.transforms/5aa75ff6087c1776d5bf80b44ea7011c/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex

Caused by: com.android.tools.r8.internal.b: Type org.nypl.simplified.profiles.controller.api.BuildConfig is defined multiple times: /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-profiles-controller-api/build/.transforms/5aa75ff6087c1776d5bf80b44ea7011c/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex, /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-deeplinks-controller-api/build/.transforms/0863ec246ae04abdc1f2d3f172457ab7/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex
```

I wonder what would cause this. I will try backing out my changes and see what specifically caused this issue.

Reply:
I backed out my changes and cleaned but it's still happening:

```
Caused by: com.android.tools.r8.internal.b: Type org.nypl.simplified.profiles.controller.api.BuildConfig is defined multiple times: /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-profiles-controller-api/build/.transforms/48b70c4be7a706d1e7c54dcdcba84dfb/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex, /Users/jdempcy/Development/ThePalaceProject/android-core/simplified-deeplinks-controller-api/build/.transforms/b0cff981bd2b792e8962aa9630843456/transformed/debug/org/nypl/simplified/profiles/controller/api/BuildConfig.dex
```

I'll refresh Gradle dependencies by right-clicking on simplified -> Refresh Gradle dependencies and see if that fixes it. I've been doing that as well, I'm not sure if it's necessary when I run Build -> Clean also, but it's worth a try.

Reply:
Ah, I found the issue. There was a mistake in the simplified-deeplinks-controller-api AndroidManifest.xml file that I just fixed!

Reply:
Ok, it's in good shape and ready for review now!